runtime/lua/vim/fs.lua

```diff
@@ -112,7 +112,11 @@ end
 ---@param ... string
 ---@return string
 function M.joinpath(...)
-  return (table.concat({ ... }, '/'):gsub('//+', '/'))
+  local path = table.concat({ ... }, '/')
+  if iswin then
+    return (path:gsub('\\', '/'):gsub('//+', '/'))
```

Comment:
nit: this makes the idea more obvious, and avoids some duplicate code

```suggestion
    path = path:gsub('\\', '/')
```